compiler/rustc_monomorphize/src/collector.rs

```diff
@@ -690,10 +661,65 @@ impl<'a, 'tcx> MirUsedCollector<'a, 'tcx> {
             return;
         }
 
+        debug!(?def_id, ?fn_span);
+
         for arg in args {
-            self.check_operand_move_size(arg, location);
+            if let Some(too_large_size) = self.operand_size_if_too_large(limit, &arg.node) {
+                self.lint_large_assignment(limit.0, too_large_size, location, arg.span);
+            };
+        }
+    }
+
+    fn operand_size_if_too_large(
+        &mut self,
+        limit: Limit,
+        operand: &mir::Operand<'tcx>,
+    ) -> Option<Size> {
```

Comment:
This doesn't check if the operand is a move operand.

Reply:
Yep, this is a known task which I added to the [tracking issue ](https://github.com/rust-lang/rust/issues/83518) a few days ago after RalfJung pointed out the same thing. I plan on fixing it relatively soon in a future PR.